cirq/ops/common_gates.py

```diff
@@ -515,6 +546,10 @@ def matrix(self):
         s = math.sqrt(0.5)
         return np.array([[s, s], [s, -s]])
 
+    def text_diagram_info(self, args: gate_features.TextDiagramInfoArgs
+                          ) -> gate_features.TextDiagramInfo:
+        return gate_features.TextDiagramInfo(('H',))
+
     def known_qasm_output(self,
```

Comment:
The known_qasm_output needs to be updated to handle the non-unit-exponent case. I think returning None should be sufficient, but you should add a test in the qasm output code using H**0.5 to be sure.